pylearner.ts

```diff
@@ -1026,6 +1026,33 @@ class SelectExpression extends Expression {
     this.push(bindingThunk(pop).value);
   }
 }
+class AppendExpression extends Expression {
+  constructor(loc: Loc, instrLoc: Loc, public target: Expression, public item: Expression) {
+    super(loc, instrLoc);
+  }
+
+  check(env: Scope) {
+    let targetType = this.target.check_(env);
+    if (!(targetType.isListType()))
+      this.executionError("Het doel van een append-uitdrukking moet een lijst zijn");
+    this.item.checkAgainst(env, intType);
+    return targetType;
+  }
+
+  async evaluate(env: Scope) {
+    await this.target.evaluate(env);
+    await this.item.evaluate(env);
+    let [target, item] = pop(2);
+    if (!(target instanceof ListObject))
+      this.executionError(target + " is geen lijst");
+    let bindingThunk = await this.target.evaluateBinding(env);
+    let listVariable = bindingThunk(pop);
+    let newElements = target.getElements();
+    newElements.push(item);
+    const newListObject = new ListObject(target.elementType, newElements);
+    this.push(listVariable.setValue(newListObject));
```

Comment:
Dit is niet hoe een `append`-oproep in Python echt werkt? In Python wijzigt een oproep `x.append(42)` het `list`-object waar `x` naar verwijst. Wat jij hier doet is een nieuw `list`-object maken en variabele `x` wijzigen zodat het naar het nieuwe `list`-object wijst. Wat jij hier schrijft wijzigt het bestaande `list`-object niet; hier is dus geen sprake van lijstmutaties, wat net het onderwerp is van jouw masterproef!

Reply:
Dit was inderdaad niet de correcte manier. Ik heb een nieuwe commit gedaan waarbij enkel het bestaande list-object gewijzigd wordt, en er geen nieuwe List-object aangemaakt moet worden. Ik heb dan wel even vastgezeten met het oplossen van een probleem rond de DOMNode en de getoonde objecten tijedns evaluatie, maar heb deze dan opgelost door een nieuwe functie.